src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/ChangeManager.java

```diff
@@ -52,21 +56,28 @@ private <P, C, S, U> boolean compareAndBuildChange(
       BiPredicate<P, C> equalityPredicate,
       BiConsumer<T, S> dataSetter) {
 
-    var convertedProposedData = proposedConverter.apply(proposedData);
-
-    if (leftOrRightDataFromPair == null && objectToAddChanges == null) {
-      return false;
+    S convertedProposedData = null;
+    if (proposedConverter != null && proposedData != null) {
+      convertedProposedData = proposedConverter.apply(proposedData);
     }
 
-    if (
-        !isEmpty(convertedProposedData) &&
-            (leftOrRightDataFromPair == null ||
-                !equalityPredicate.test(proposedData,
-                    currentDataGetter.apply(leftOrRightDataFromPair)))
-    ) {
-      dataSetter.accept(objectToAddChanges, convertedProposedData);
-      return true;
+    if (!isEmpty(convertedProposedData)) {
+      C currentData = null;
+      if (currentDataGetter != null && leftOrRightDataFromPair != null) {
+        currentData = currentDataGetter.apply(leftOrRightDataFromPair);
+      }
+
+      if (leftOrRightDataFromPair == null || (equalityPredicate != null && !equalityPredicate.test(
+          proposedData, currentData))) {
```

Comment:
Should we be comparing the convertedProposedData to the currentData? 
proposedData is in terms of BO/MOP model while current data and changed/setted are in terms of PCSs/Officers?
Or are we comparing apples to oranges?
If not then only need to convert if setting so something does not seem to line up here.

Reply:
So the converter converts the data found in the public/private data into something that will be accepted in the model

The equalityPredicate checks to see if the data from the proposedData (i.e. The submission DTOs) is the same as that is the public/private data 

It is possible and I'm pretty sure does happen a number of times that the proposedData DataType is different to both the currentData DataType and the Setter DataType 

Hence why the converted data isn't used here